libbpfgo/libbpfgo.go

```diff
@@ -478,6 +478,65 @@ func (b *BPFMap) Update(key, value interface{}) error {
 	return nil
 }
 
+type BPFMapIterator struct {
+	b *BPFMap
+
+	err error
+
+	keySize int
+
+	prev []byte
+	next []byte
+}
+
+func (b *BPFMap) Iter(keySize int) *BPFMapIterator {
+	return &BPFMapIterator{
+		b:       b,
+		keySize: keySize,
+		prev:    nil,
+		next:    nil,
+	}
+}
+
+func (it *BPFMapIterator) Next() bool {
+	if it.err != nil {
+		return false
+	}
+
+	prevPtr := unsafe.Pointer(nil)
+	if it.next != nil {
+		prevPtr = unsafe.Pointer(&it.next[0])
+	}
+
+	next := make([]byte, it.keySize)
+	nextPtr := unsafe.Pointer(&next[0])
+
+	errC, err := C.bpf_map_get_next_key(it.b.fd, prevPtr, nextPtr)
```

Comment:
How does have two return values work with CGO calls work?

Reply:
Excerpt from the cgo docs:

> Any C function (even void functions) may be called in a multiple assignment context to retrieve both the return value (if any) and the C errno variable as an error (use _ to skip the result value if the function returns void). For example:
>
> n, err = C.sqrt(-1)
> _, err := C.voidFunc()
> var n, err = C.sqrt(1)